numba/targets/arraymath.py

```diff
@@ -668,6 +668,101 @@ def _trivial_isnan(x):
         return _trivial_isnan
 
 
+@overload(np.iscomplex)
+def np_iscomplex(x):
+    def impl(x):
+        ax = np.asarray(x) # NumPy uses asanyarray here!
```

Comment:
This should not return False for types that cannot `type_can_asarray`, i.e. `np.iscomplex('string')`.

Reply:
Hum, not sure if I followed it right. You're saying that the output of `np.iscomplex('string') **should not** be False?

Reply:
I'm trying to express that if a call to `type_can_asarray` on the argument returns `False` then this function should just return `False`.